pkg/apis/cnsoperator/config/cnsregistervolume_crd.yaml

```diff
@@ -45,6 +47,7 @@ spec:
               description: DiskUrlPath is URL path to an existing block volume to be
                 imported into Project Pacific cluster.
               type: string
+              pattern: '^(http[s]?:\/\/)?([^\/\s]+\/folder\/)(.*)$'
```

Comment:
Can we add validation for accessMode?

Reply:
We don't want to do that as these values are of type v1.PersistentVolumeAccessMode and shouldn't be hardcoded as patterns in yaml. 

We already have such validations for these in controller.

Reply:
What happens if I set accessMode as "abc"? Would the CR creation fail?

Reply:
No, CRD will be created but reconciler will mark this failed.

Reply:
Isn't that too late? Ideally we should reject invalid CRs before even storing it in API server. Can we use `enum` to validate the allowed strings?

Reply:
I don't want to hardcode strings here. What if PersistentVolumeAccessMode adds new type, we should modify our YAMLs for it. So, I decided to add that as part of code only.

Reply:
What happens when the code fails, does it get invoked again and again with the invalid filed? Would that mean the controller keeps continuously failing forever? If yes, thats a problem since this could be avoided by proper validation in the first place to reject CR creation.